lib/application.js

```diff
@@ -102,6 +103,9 @@ module.exports = class Application extends Emitter {
    */
 
   use(fn) {
+    if (this.forceAsync && fn.constructor.name !== 'AsyncFunction') {
+      throw new TypeError('middleware must be an AsyncFunction!');
+    }
```

Comment:
Ah, you're right. For that reason I agree that it's better to reorder the checks to avoid duplicating errors/checks. :) But again, holding off a bit until discussion settles.